linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/FunctionEmitter.scala

```diff
@@ -1920,7 +2079,11 @@ private class FunctionEmitter private (
         fb += wa.GlobalGet(genGlobalID.undef)
 
       case StringType =>
-        fb += wa.RefAsNonNull
+        val sig = watpe.FunctionType(List(watpe.RefType.anyref), List(watpe.RefType.any))
+        fb.block(sig) { nonNullLabel =>
+          fb += wa.BrOnNonNull(nonNullLabel)
+          fb += wa.GlobalGet(genGlobalID.emptyString)
+        }
```

Comment:
Hmm... This was a bug, no? If we didn't catch this, doesn't it point to a weakness in the test suite?

Also, should probably be separate?

Reply:
Indeed, that was a bug, although a theoretical one. We have source syntax to emit an `asInstanceOf[string]`, so it is not user-visible.

I did separate it into a commit of its own.